src/processor/operator/order_by/sort_state.cpp

```diff
@@ -107,53 +108,85 @@ PayloadScanner::PayloadScanner(MergedKeyBlocks* keyBlockToScan,
 }
 
 uint64_t PayloadScanner::scan(std::vector<common::ValueVector*> vectorsToRead) {
-    if (nextTupleIdxToReadInMergedKeyBlock >= endTuplesIdxToReadInMergedKeyBlock) {
+    if (limitNumber <= 0 ||
```

Comment:
Do we have a test where limitNumber < 0 ? I thought binder will throw in this case.

Reply:
This variable is used as a counter, so i will decrease the counter after each scan.